### ISSUE TITLE
[REVIEW]: Alarconpy: a Python Package for Meteorologists

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2479]
**Submitting author:** @apalarcon (<a href="http://orcid.org/0000-0002-9454-2331">Albenis Pérez Alarcón</a>)
**Repository:** <a href="https://github.com/apalarcon/alarconpy/" target ="_blank">https://github.com/apalarcon/alarconpy/</a>
**Version:** v1.0.3
**Editor:** @dhhagan
**Reviewers:** @hertzsprung, @kgoebber
**Archive:** <a href="https://doi.org/10.5281/zenodo.4493257" target="_blank">10.5281/zenodo.4493257</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/d134560d225371f6b159cfdaa01c7fd4/status.svg)](https://joss.theoj.org/papers/d134560d225371f6b159cfdaa01c7fd4)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/d134560d225371f6b159cfdaa01c7fd4"><img src="https://joss.theoj.org/papers/d134560d225371f6b159cfdaa01c7fd4/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/d134560d225371f6b159cfdaa01c7fd4/status.svg)](https://joss.theoj.org/papers/d134560d225371f6b159cfdaa01c7fd4)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/apalarcon/alarconpy/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@kgoebber & @hertzsprung, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @dhhagan know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/apalarcon/alarconpy/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@apalarcon) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/apalarcon/alarconpy/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@apalarcon) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @RCHG, @hertzsprung it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.21105/jose.00037 is OK
- 10.5065/D6WW7G29 is OK
- 10.5065/D6CN72NW is OK
- 10.1109/MCSE.2007.53 is OK

MISSING DOIs

- https://doi.org/10.1109/tools.1997.654712 may be missing for title: Steering Object-Oriented Scientific Computations

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02479/joss.02479/10.21105.joss.02479.pdf)

**Comment by RCHG** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by dhhagan** [comment]
Hi @RCHG - you should be able to edit the issue once you accept the invite. If you did not get an invite, please let me know and I will re-invite you to the issue. As for (b) - yes, please indicate the steps right here in this issue. Please let me know if you have any other qustions!

**Comment by RCHG** [comment]
Hi @dhhagan - I am not aware that I had an invitation.

**Comment by hertzsprung** [comment]
@dhhagan In case you missed it, I emailed you with a query about this review.  Do let me know if you need me to resend it.

**Comment by dhhagan** [comment]
@whedon re-invite @RCHG as reviewer

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@rchg please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by dhhagan** [comment]
Hi @hertzsprung - please resend! I haven't seen it yet, but am happy to answer any questions. Feel free to ask here in the comments as well if you feel comfortable doing so.

**Comment by dhhagan** [comment]
@RCHG were you able to accept the invite?

**Comment by hertzsprung** [comment]
In my view, Alarconpy fails to represent a substantial scholarly effort because it comprises numerous utility functions that are already provided by existing libraries, including:

* interpolation functions already provided by scipy
* statistics functions and haversine function already provided by scikit-learn
* unit conversion already provided by [Unum](https://github.com/trzemecki/Unum)

It would be more appropriate for any missing features to be added to these existing libraries.  Some of Alarconpy's thermodynamic functions might find a good home in [metpy](https://unidata.github.io/MetPy/), which is already a project dependency.

**Comment by dhhagan** [comment]
@whedon remind @RCHG in 1 day

**Comment by whedon** [comment]
Reminder set for @RCHG in 1 day

**Comment by whedon** [comment]
:wave: @RCHG, please update us on how your review is going.

**Comment by dhhagan** [comment]
@whedon remove @RCHG as reviewer

**Comment by whedon** [comment]
OK, @RCHG is no longer a reviewer

**Comment by dhhagan** [comment]
@Mankoff @jsta @sgrieve are any of you able to (and interested in ) reviewing this paper?

**Comment by apalarcon** [comment]
I suggest this reviewer Roberto Carlos Cruz Rodríguez, roberto.cruz.rdg@gmail.com

**Comment by sgrieve** [comment]
I'm afraid I don't have any capacity to take on a review for the next few weeks at least.

**Comment by jsta** [comment]
I too am unable to take on a review at this time.

**Comment by apalarcon** [comment]
I suggest this reviewer Oscar Díaz Rodríguez oodr72@gmail.com

**Comment by dhhagan** [comment]
Hey @apalarcon I am waiting to hear back from a few others invited via email and will invite Oscar as well if needed. Thanks!

**Comment by apalarcon** [comment]
Hi @dhhagan 
I would like to know the status of the review.

**Comment by dhhagan** [comment]
Sorry for the very long delay, @apalarcon I still have not been able to identify a second reviewer. I am still waiting to hear back from a couple and will ping them again right now offline.

**Comment by apalarcon** [comment]
I further suggest these reviewers
Maibys Sierra Lorenzo maibyssl@gmail.com
Adrian Luis Ferrer Hernández adrian.ferrer@insmet.cu

**Comment by apalarcon** [comment]
Hi! @dhhagan, do you have any news about the review process?

**Comment by dhhagan** [comment]
@apalarcon Finally got the second reviewer (@kgoebber) - sorry for the long delay.

**Comment by dhhagan** [comment]
@whedon re-invite @kgoebber as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@kgoebber please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by kgoebber** [comment]
**General Comments**

Overall this package provides a mechanism to facilitate the inspection of various meteorological data. Most advanced functionality in this package comes from other well-established Python modules including SciPy and MetPy. Overall, it is hard to see extensive use of this package by the broader community due its limited scope, scant documentation, and better offering of similar functionality in other Python libraries.

**Specific Comments**
Substantial scholarly effort: This package does not meet the criteria for acceptance. It is a young project with a relatively few number of commits. This effort is focused on the main authors work flow and has a couple of features not present in other modules, but would be better served by inclusion in those packages.

Installation: Installation went fine, but it was confusing which folder to copy as there as both the main repository folder and the source code folder are named the same. Dependency installation is left to the user, which is not as problematic in this instance due to all required packages are easy installed through package management systems.

Functionality: In general, what I tested worked fine, but it was hard to determine the full extent of the functionality through the limited documentation and examples given in the repository.

A statement of need: While there was some basic statement of how this package might be useful to others, the scope of the project is not beyond that for which other packages already exist within the domain space. For example, the plotting capabilities of this repository make getting basic plots relatively easy, there is more robust functionality for similar plots already in MetPy that make full use of that library.

Example usage:  There are a number of examples, however, some of them are incomplete. For example, the interpolation example leaves out the important data input lines to have a clear example of what data are going into the interpolation routines.

Automated tests: No tests are conducted as a part of this package to verify accuracy and consistency as changes are made.

Community guidelines: There are no guidelines for being able to contribute to this package, however, there is information on who to contact, via email, about the package and its operation.

Software paper: The paper fulfills the basic requirements set forth in a statement of need and context of this work within the Python ecosystem.

**Comment by dhhagan** [comment]
@apalarcon @hertzsprung @kgoebber @openjournals/jose-eics 

First, I would like to thank you (@apalarcon) for submitting to JOSS. I hope that, while delayed, this process was useful and helpful for you and your work. Upon review, both reviewers have recommended this manuscript for rejection from JOSS as it currently stands due to the fact it mostly represents smaller utility functions than/should be incorporated into existing packages where they aren't already and the supporting work around the main code needs to be substantially improved (see specific comments from @kgoebber above). Thank you very much for taking the time to submit and go through the review process with JOSS - I will let the @openjournals/jose-eics take it over from here. Please reach out if you have any questions, comments, or concerns.

**Comment by labarba** [comment]
ping @openjournals/joss-eics

**Comment by dhhagan** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by kyleniemeyer** [comment]
Hi @apalarcon, can you please archive your software repo on Zenodo/Figshare, and provide the DOi here?

cc @dhhagan

**Comment by apalarcon** [comment]
Hi @kyleniemeyer  

Here de DOI in Zenodo 
http://doi.org/10.5281/zenodo.4493257

**Comment by kyleniemeyer** [comment]
@whedon set 10.5281/zenodo.4493257 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4493257" target="_blank">10.5281/zenodo.4493257</a> is the archive.

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/jose.00037 is OK
- 10.5065/D6WW7G29 is OK
- 10.5065/D6CN72NW is OK
- 10.1109/MCSE.2007.53 is OK

MISSING DOIs

- 10.1109/tools.1997.654712 may be a valid DOI for title: Steering Object-Oriented Scientific Computations

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2069

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2069, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
Hi @apalarcon, my apologies but I need one more edit to the paper from you: can you please add a short Statement of Need section? This is part of our [required elements](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) for a JOSS paper. You may be able to move some of the text in your summary. 

There is no need to re-deposit the software in Zenodo, since JOSS archives the paper text.

(Normally whedon checks for this section upon submission, I'm not sure how we missed this.)

**Comment by apalarcon** [comment]
Hi @kyleniemeyer  

Done!

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02479/joss.02479/10.21105.joss.02479.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02479/joss.02479/10.21105.joss.02479.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1356731476398780417) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2070
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02479](https://doi.org/10.21105/joss.02479)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats @apalarcon on your article's publication in JOSS!

Many thanks to  @hertzsprung and @kgoebber for reviewing this, and @dhhagan for editing it.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02479/status.svg)](https://doi.org/10.21105/joss.02479)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02479">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02479/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02479/status.svg
   :target: https://doi.org/10.21105/joss.02479
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02479/status.svg)](https://doi.org/10.21105/joss.02479)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by apalarcon** [comment]
Thanks

**Comment by arfon** [comment]
Hi all, it looks like a mistake was made here in the handoff between editors on rotation this week. For context, it looks like the following happened:

- @dhhagan (the handling editor) reminded me on Slack about this paper. I failed to notice that this was a paper that we had been discussing before and so asked him to run `@whedon accept` which he did [here](https://github.com/openjournals/joss-reviews/issues/2479#issuecomment-770405526). This is how editors typically notify the EiC team that the paper is ready to publish and accept (which is incorrect in this case).
- The newly incoming EiC on rotation appears to have noticed the `@whedon accept`, and presumably thinking the paper was in good shape to move forward, asked the author to make an archive of their software so that they could continue the acceptance process.

Based on the thorough feedback of the reviewers, our decision here was actually to _reject_ this submission as it fails to meet our requirements for substantial scholarly effort.

I'll work with the editorial team to work out what our next steps here and report back later today on how we plan to handle this submission.

Apologies to all for any confusion and frustration caused. This looks to simply be an error on our part.

**Comment by arfon** [comment]
Hi all, just a follow-up on my earlier message here. I've confirmed the above events with the EiC team.

As the acceptance here was processed in error, we've contacted Crossref (the DOI registering service for JOSS) who have deleted the DOI associated with this paper (https://doi.org/10.21105/joss.02479). 

@apalarcon, @dhhagan, @hertzsprung, @kgoebber - my sincere apologies for any confusion caused here. We will use this incident to learn and improve our processes.

@apalarcon - based on the feedback of both reviewers and in discussions with the handling editor here, the decision is to reject this submission as it fails to meet our requirements for substantial scholarly effort.

Re-submissions are welcome in JOSS, but as @dhhagan mentioned outlined above, this submission would require significant work to be considered publishable here.
